tests/node-18.0/tests.js

```diff
@@ -1,37 +1,71 @@
 const fetch = require("node-fetch");
 
-/*
-    'req' variable has:
-        'headers' - object with request headers
-        'payload' - object with request body data
-        'variables' - object with function variables
-    'res' variable has:
-        'send(text, status)' - function to return text response. Status code defaults to 200
-        'json(obj, status)' - function to return JSON response. Status code defaults to 200
-    
-    If an error is thrown, a response with code 500 will be returned.
-*/
+module.exports = async (context) => {
+    const action = context.req.headers['x-action'];
 
-module.exports = async (req, res) => {
-    const payload = JSON.parse(req.payload === '' ? '{}' : req.payload);
+    switch (action) {
+        case 'plaintextResponse':
+            return context.res.send('Hello World 👋');
+        case 'jsonResponse':
+            return context.res.json({ json: true, message: 'Developers are awesome.' });
+        case 'fileResponse':
+            return context.res.file(__dirname + '/resources/picture.png', undefined, { 'content-type': 'image/png' });
+        case 'redirectResponse':
+            return context.res.redirect('https://github.com/');
+        case 'emptyResponse':
+            return context.res.empty();
+        case 'noResponse':
+            context.res.send('This should be ignored, as it is not returned.');
+            break;
+        case 'doubleResponse':
+            context.res.send('This should be ignored.');
+            return context.res.send('This should be returned.');
+        case 'headersResponse':
+            return context.res.send('OK', 200, { 'first-header': 'first-value', 'second-header': 'second-value' });
+        case 'statusResponse':
+            return context.res.send('FAIL', 404);
+        case 'complexResponse':
+            context.res.setCode(201);
+            context.res.setHeaders({ 'header1': 'value1' })
+            context.res.addHeader('header2', 'value2');
+            context.res.setBody('👌');
+            context.res.setContentType('application/json');
+            context.res.addCookie('cookie1', 'value1');
+            context.res.addCookie('cookie2', 'value2', 'Wed, 21 Oct 2015 07:28:00 GMT', 2592000, '/', 'google.com', true, true, 'Lax');
```

Comment:
With this approach, `context.res` serves two separate purposes, I think there could be two separate types: one that lets you send a response (has `send`, `json`, `redirect`, etc.) and one that lets you build a response _to_ send (has `setStatusCode`, `setHeaders`, `setBody`, etc. )